src/main/java/biz/netcentric/maven/extension/repofromenv/FromEnvReposConfigurationProcessor.java

```diff
@@ -41,47 +43,80 @@
-
-        List<RepoFromEnv> reposFromEnv = getReposFromEnv(System.getenv());
+        
+        Map<String, String> sysEnv = System.getenv();
+        isVerbose = Boolean.valueOf(sysEnv.get(ENV_PROP_MVN_SETTINGS_REPO_VERBOSE));
+        
+        List<RepoFromEnv> reposFromEnv = getReposFromEnv(sysEnv);
 
         configureMavenExecution(cliRequest.getRequest(), reposFromEnv);
-        
 
     }
 
     void configureMavenExecution(MavenExecutionRequest request, List<RepoFromEnv> reposFromEnv) {
         if (!reposFromEnv.isEmpty()) {
+            
+            logRepositoriesAndMirrors(request);
+            
             Profile repositoriesFromEnv = new Profile();
     
             for (RepoFromEnv repoFromEnv : reposFromEnv) {
     
-                Repository repository = getRepository(repoFromEnv);
+                repositoriesFromEnv.addRepository(getRepository(repoFromEnv));
+                
+                repositoriesFromEnv.addPluginRepository(getRepository(repoFromEnv));
     
-                repositoriesFromEnv.addRepository(repository);
-    
-                if (repoFromEnv.getUsername() != null && repoFromEnv.getPassword() != null) {
-                    Server server = new Server();
-                    server.setId(repoFromEnv.getId());
-                    server.setUsername(repoFromEnv.getUsername());
-                    server.setPassword(repoFromEnv.getPassword());
-                    request.addServer(server);
-                }
-                logger.debug("Repository from system env: " + repoFromEnv.getUrl() + " (id: "+repoFromEnv.getId() 
+                if (repoFromEnv.getUsername() != null) {
+                    request.addServer(getServer(repoFromEnv));
+                } 
+                
+                // minimal line that we always log (regardless of MVN_SETTINGS_REPO_LOG_VERBOSE or -X parameter)
+                logger.info("Repository from system env: " + repoFromEnv.getUrl() + " (id: "+repoFromEnv.getId() 
                         + (repoFromEnv.getUsername() != null ? " user: " + repoFromEnv.getUsername(): "") + ")");
             }
     
             // activate profile
             repositoriesFromEnv.setId(PROFILE_ID_REPOSITORIES_FROM_ENV);
             request.addProfile(repositoriesFromEnv);
             request.addActiveProfile(PROFILE_ID_REPOSITORIES_FROM_ENV);
+            
         }
     }
 
+    private void logVerbose(String msg) {
+        if(isVerbose) {
+            logger.info(msg);
+        } else {
+            logger.debug(msg);
+        }
+    }
+
+    private void logRepositoriesAndMirrors(MavenExecutionRequest request) {
+
+        List<ArtifactRepository> repositories = request.getRemoteRepositories();
+        List<ArtifactRepository> pluginRepositories = request.getPluginArtifactRepositories();
+        List<Mirror> mirrors = request.getMirrors();
+        logVerbose("Configured in settings.xml: repositores: "+
```

Comment:
Not necessarily only in settings.xml, right? Doesn't this also expose the repos from pom.xml?

Reply:
At this time of the lifecycle you only get a hold of what is configured in settings.xml